gateway/ln-gateway/src/lib.rs

```diff
@@ -1118,6 +1138,69 @@ impl Gateway {
             config,
         }
     }
+
+    async fn check_federation_network(
+        &self,
+        info: &FederationInfo,
+        network: Network,
+    ) -> Result<()> {
+        debug!("Checking federation network");
+        debug!(?info, "Federation info");
+        debug!(?network, "Gateway network");
+        // WIP: Check wallet module for configured network
```

Comment:
wondering how we can best check federation network. ref: https://github.com/fedimint/fedimint/issues/3450#issuecomment-1778903612